Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -328,596 +365,450 @@ public virtual bool HandlesOnSpeech(Mobile from)
 
     public virtual void OnSpeech(SpeechEventArgs e)
     {
-        if (e.Mobile.Alive && e.Mobile.InRange(m_Mobile.Location, 3) && m_Mobile.IsHumanInTown())
+        if (WasNamed(e.Speech) && e.Mobile.Alive && 
+            e.Mobile.InRange(m_Mobile.Location, 3) && m_Mobile.IsHumanInTown())
         {
-            if (e.HasKeyword(0x9D) && WasNamed(e.Speech)) // *move*
+            if (e.HasKeyword(0x9D)) // *move*
             {
-                if (m_Mobile.Combatant != null)
-                {
-                    // I am too busy fighting to deal with thee!
-                    m_Mobile.PublicOverheadMessage(MessageType.Regular, 0x3B2, 501482);
-                }
-                else
-                {
-                    // Excuse me?
-                    m_Mobile.PublicOverheadMessage(MessageType.Regular, 0x3B2, 501516);
-                    WalkRandomInHome(2, 2, 1);
-                }
+                m_Mobile.PublicOverheadMessage(MessageType.Regular, 0x3B2, 501516);
+                // 501516: Excuse me?
+                m_Mobile.Location = new Point3D(m_Mobile.Location.X + 
+                Utility.RandomMinMax(-1, 1), m_Mobile.Location.Y + 
+                Utility.RandomMinMax(-1, 1), m_Mobile.Location.Z);
```

Comment:
Why are we force moving them, potentially into a wall or obstruction, before we make them walk?